pkg/services/ngalert/state/state_tracker.go

```diff
@@ -53,22 +54,36 @@ func NewStateTracker(logger log.Logger) *StateTracker {
 	return tracker
 }
 
-func (st *StateTracker) getOrCreate(uid string, orgId int64, result eval.Result) AlertState {
+func (st *StateTracker) getOrCreate(alertRule *ngModels.AlertRule, result eval.Result) AlertState {
 	st.stateCache.mu.Lock()
 	defer st.stateCache.mu.Unlock()
+	lbs := alertRule.Labels
+	if len(result.Instance) > 0 {
+		lbs = result.Instance
+	}
+	lbs["__alert_rule_uid__"] = alertRule.UID
```

Comment:
We'll need to make these labels public constants, as the notification system should be able to interpret them and either rename them or take them into account for other things.

Reply:
Sounds good. Definitely threw a few things in there for the purpose of demoing